src/Symfony/Component/Routing/Generator/Dumper/CompiledUrlGeneratorDumper.php

```diff
@@ -35,12 +37,69 @@ public function getCompiledRoutes(): array
                 $compiledRoute->getTokens(),
                 $compiledRoute->getHostTokens(),
                 $route->getSchemes(),
+                [],
             ];
         }
 
         return $compiledRoutes;
     }
 
+    public function getCompiledAliases(): array
+    {
+        $routes = $this->getRoutes();
+        $compiledAliases = [];
+        foreach ($routes->getAliases() as $name => $alias) {
+            $deprecations = [];
+
+            if ($alias->isDeprecated()) {
+                $deprecation = $alias->getDeprecation($name);
+
+                $deprecations[] = $deprecation;
+
+                trigger_deprecation($deprecation['package'], $deprecation['version'], $deprecation['message']);
```

Comment:
This is indeed wrong. Dumping the URL generator should not trigger a deprecation. It should only be triggered when generating the url.
The only compile-time deprecation should be when a non-deprecated alias targets a deprecated one.